tests/lib.rs

```diff
@@ -46,8 +46,8 @@ fn data_map_content_only(){
   assert!(dm.has_chunks() == false);
 }
 
-fn random_string(length: u64) -> String {
-  (0..length).map(|_| (0x20u8 + (rand::random::<f32>() * 96.0) as u8) as char).collect()
+fn random_bytes(length: u64) -> Vec<u8> {
+  (0..length).map(|_| (0x20u8 + (rand::random::<f32>() * 96.0) as u8) as u8).collect()
```

Comment:
I think there needs to be some consideration, I'm assuming the system allows files > 4gb, so there needs to be a mechanism for loading these files on a 32bit machine, using usize/vec will not allow this afaik.

But yeah, positions/lengths should be usize as that's equivalent to `size_t` 'guaranteed to be big enough for the size of the biggest object your system can handle'.